ice.go

```diff
@@ -141,3 +160,87 @@ func (c *ICECandidate) Unmarshal(raw string) error {
 
 	return nil
 }
+
+// GetICEParams extracts session-level ICE parameters
+func (s *SessionDescription) GetICEParams() ICESessionParameters {
+	ret := ICESessionParameters{}
+
+	var value string
```

Comment:
I would just scope each value/found to conditionals. Not a hot path so no worries about optimizing/reuse!